src/main/scala/com.snowplowanalytics.iglu/ctl/commands/TableMigrate.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2012-2019 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.iglu.ctl.commands
+
+import java.nio.file.Path
+
+import cats.data._
+import cats.effect._
+import cats.implicits._
+import fs2.Stream
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaMap}
+import com.snowplowanalytics.iglu.schemaddl.{StringUtils => SchemaDDLStringUtils}
+import com.snowplowanalytics.iglu.schemaddl.redshift.{Column => DDLColumn}
+import com.snowplowanalytics.iglu.ctl.Command.DbConfig
+import com.snowplowanalytics.iglu.ctl.commands.TableCheck.TableCheckResult
+import com.snowplowanalytics.iglu.ctl.commands.TableCheck.TableCheckResult._
+import com.snowplowanalytics.iglu.ctl.{Common, Failing, Result, Storage}
+
+object TableMigrate {
+
+  val EventFieldSeparator = "\\t"
+
+  /**
+    * Primary method of table-migrate command. Checks whether table of the given schema
+    * has expected structure which is determined with last and previous versions of
+    * the given schema and if it has not expected structure, creates necessary migration
+    * SQL statements
+    */
+  def process(resolver: Path,
+              schema: SchemaKey,
+              dbSchema: String,
+              outputS3Path: String,
+              awsRole: String,
+              awsRegion: String,
+              useGzip: Boolean,
+              dbConfig: DbConfig)(implicit cs: ContextShift[IO], t: Timer[IO]): Result = {
+    val stream = for {
```

Comment:
It seems that `Stream[IO, A]` is an unnecessary abstraction here. Instead this can be just `IO[A]`. We don't use any `Stream`-specific features here (parallelism, resource-safety, multiple result emitting etc). To transform `Resource` into `IO` you can use `resource.use { r => ??? }`